### ISSUE TITLE
Paste limited to 50 chars when run in Docker container on Windows

### ISSUE THREAD

**Issue by dhirschfeld** [Microsoft/console#38]
When pasting text into powershell in a Windows container the text is truncated at 50 characters which makes for a very frustrating and unproductive user experience. 

This effect is most notable in powershell becaue of an unfortuante interaction with  PSReadline however [unusual/undersirable behaviour](https://github.com/lzybkr/PSReadLine/issues/460#issuecomment-345609929) is also observed in a plain `cmd` terminal leading to (IIUC) the conclusion that the underlying problem is with the console itself.

This issue has been previously opened in 3 separate repositories:

 * lzybkr/PSReadLine#460
 * moby/moby#27793
 * PowerShell/PowerShell#5501

...however I'm opening it here as all those issues have been closed under the assumption that the problem lies with the console itself

**Comment by dhirschfeld** [comment]
In the PSReadline issue @jstarks [commented](https://github.com/lzybkr/PSReadLine/issues/460#issuecomment-341729742):
> *Agreed that this is annoying. We're hoping to improve the console API compatibility in `windowsservercore` in the next Windows release, which I suspect will resolve this and several other issues. Currently I don't have any plans to debug this with the current set of images, unfortunately.*

My main goal in opening the issue here it so I can follow any progress in fixing the problem, and in particular if there is any estimated timeline for a fix being released.

**Comment by zadjii-msft** [comment]
This is technically an issue with Docker on Windows, but yes, we're working on a solution.

**Comment by flupkede** [comment]
It's just amazing that something as trivial as this is simply not working.

**Comment by cblackuk** [comment]
@zadjii-msft Cannot wait for this to be fixed... it is so annoying... Any workaround in the interim?

**Comment by dhirschfeld** [comment]
See https://github.com/lzybkr/PSReadLine/issues/460#issuecomment-341733715 for a workaround.

**Comment by cblackuk** [comment]
@dhirschfeld Saved my life! Thanks a lot!

**Comment by cblackuk** [comment]
@dhirschfeld This allows me to paste fine but now you cannot go back a character or word it just deletes everything if you want to edit whatever you typed or pasted. Any workaround for that?

**Comment by zadjii-msft** [comment]
I'll defer back to @jstarks who is more familiar with Windows containers and might be able to share more information on the subject.

**Comment by DougPlumley** [comment]
No updates on this huh?  Still seems to be an issue.

**Comment by zadjii-msft** [comment]
Actually @jstarks might be able to comment on this. There should be a fix going in to CExecSvc (which powers docker server on windows) to fix this, but I'm not really sure about their release schedule.

**Comment by jsturtevant** [comment]
This is fixed on the Windows 2019 preview image.  You can try it out on azure:

![image](https://user-images.githubusercontent.com/648372/43410839-226cf66a-93dd-11e8-8d96-4593721026a2.png)

**Comment by dhirschfeld** [comment]
Will that fix make it to `1803` or  `1809`?

**Comment by malcolm-boden** [comment]
Is there any of this working on 1607 ?

**Comment by zadjii-msft** [comment]
@malcolm-boden Nope, conpty support was only added in 1809 (October 2018 release), so you'll have to upgrade to a newer version to get the fix.

**Comment by robe070** [comment]
Still in 1809 👎

**Comment by triynko** [comment]
2021, latest version of docker and CANNOT PASTE A SINGLE FULL COMMAND INTO DOCKER. Where did this 50 character limit come from? It makes no sense.

**Comment by DHowett** [comment]
We’ve pretty well proven that the issue isn’t in this repository or in the console or the terminal, so somebody is going to have to report the issue wherever it truly belongs.
